core/src/main/java/org/testcontainers/utility/LogUtils.java

```diff
@@ -19,15 +18,81 @@
  */
 @UtilityClass
 public class LogUtils {
+
     /**
-     * {@inheritDoc}
+     * Attach a log consumer to a container's log outputs in follow mode. The consumer will receive all previous
+     * and all future log frames of the specified type(s).
+     *
+     * @param dockerClient a Docker client
+     * @param containerId  container ID to attach to
+     * @param consumer     a consumer of {@link OutputFrame}s
+     * @param types        types of {@link OutputFrame} to receive
      */
-    public void followOutput(DockerClient dockerClient, String containerId,
-                             Consumer<OutputFrame> consumer, OutputFrame.OutputType... types) {
+    public void followOutput(DockerClient dockerClient,
+                             String containerId,
+                             Consumer<OutputFrame> consumer,
+                             OutputFrame.OutputType... types) {
+
+        attachConsumer(dockerClient, containerId, consumer, true, types);
+    }
+
+    /**
+     * Attach a log consumer to a container's log outputs in follow mode. The consumer will receive all previous
+     * and all future log frames (both stdout and stderr).
+     *
+     * @param dockerClient a Docker client
+     * @param containerId  container ID to attach to
+     * @param consumer     a consumer of {@link OutputFrame}s
+     */
+    public void followOutput(DockerClient dockerClient,
+                             String containerId,
+                             Consumer<OutputFrame> consumer) {
+
+        followOutput(dockerClient, containerId, consumer, STDOUT, STDERR);
+    }
+
+    /**
+     * Retrieve all previous log outputs for a container of the specified type(s).
+     *
+     * @param dockerClient a Docker client
+     * @param containerId  container ID to attach to
+     * @param types        types of {@link OutputFrame} to receive
+     * @return all previous output frames (stdout/stderr being separated by newline characters)
+     */
+    public String getOutput(DockerClient dockerClient,
+                            String containerId,
+                            OutputFrame.OutputType... types) {
+
+        final ToStringConsumer consumer = new ToStringConsumer();
+        final WaitingConsumer wait = new WaitingConsumer();
+        attachConsumer(dockerClient, containerId, consumer.andThen(wait), false, types);
+
+        wait.waitUntilEnd();
+        return consumer.toUtf8String();
+    }
+
+    /**
+     * Retrieve all previous log outputs for a container (both stdout and stderr)
+     *
+     * @param dockerClient a Docker client
+     * @param containerId  container ID to attach to
+     * @return all previous output frames (stdout/stderr being separated by newline characters)
+     */
+    public String getOutput(DockerClient dockerClient,
```

Comment:
this additional method could be avoided because it can be handled in another `getOutput` (`types.length == 0` case), WDYT?